pyelftool_parser/src/analyzer.py

```diff
@@ -0,0 +1,203 @@
+import sys
+import register
+import execute
+from debug import loginst, log, logresult, logstack
+from capstone.x86 import *
+from elftools.elf.elffile import ELFFile
+from capstone import *
+from elftools.elf.sections import (
+    NoteSection, SymbolTableSection, SymbolTableIndexSection
+)
+class disassembler:
+    def __init__(self, path):
+        self.path = path
+        self.inst = dict()
+        self.symbol = dict()
+        self.vertex = dict()
+        self.entry_pc = 0
+        self.exit_pc = 0
+        self.acquire_stack_address = 0
+
+    def disasminst(self):
+        pc_flag = 0
+        with open(self.path, 'rb') as f:
+            elf = ELFFile(f)
+            code = elf.get_section_by_name('.text')
+            ops = code.data()
+            addr = code['sh_addr']
+            md = Cs(CS_ARCH_X86, CS_MODE_64)
+            md.detail = True
+            for i in md.disasm(ops, addr):
+                self.inst[i.address] = (i)
+                if (i.address in self.symbol):  ## it is to catch the next symbol start. @minghwu: I think it could have a better way to do this.
+                    pc_flag = 0
+                if (i.address == self.entry_pc): ## when it is entry point, set the flag = 1 to catch exitpoint.
+                    pc_flag = 1
+                if (pc_flag == 1):               ## catch the point until the next symbol, means it is exit point.
+                    self.exit_pc = i.address
+                log(f'0x{i.address:x}:\t{i.mnemonic}\t{i.op_str}')
```

Comment:
I haven't run this, but does this log quite a bit to stdout/err? We'd like the output to be relatively quiet in most cases.

Reply:
Yes, there is a debug.py. It is actually a log system. We could decide what level log to stdout.